src/Transition.js

```diff
@@ -44,51 +46,41 @@ class Transition extends React.Component {
   }
 
   componentWillReceiveProps(nextProps) {
-    if (nextProps.in && this.props.unmountOnExit) {
-      if (this.state.status === UNMOUNTED) {
-        // Start enter transition in componentDidUpdate.
-        this.setState({status: EXITED});
-      }
-    }
-    else {
-      this._needsUpdate = true
+    this._propsUpdated = true
+
+    if (nextProps.in && this.state.status === UNMOUNTED) {
+      // Start enter transition in componentDidUpdate.
+      this.setState({ status: EXITED });
```

Comment:
just for clarity i liked not setting `_propsUpdated` on this path, since the transitional update logic will handle this